File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque

N, M, V = map(int, input().split())

graph = [[] for _ in range(N + 1)]
for i in range(M):
    x, y = map(int, input().split())
    graph[x].append(y)
    graph[y].append(x)
    graph[x].sort()
    graph[y].sort()

visited1 = [False] * (N + 1)
visited2 = [False] * (N + 1)

def DFS(graph, v, visited):
    visited[v] = True
    print(v, end=' ')
    for i in graph[v]:
        if not visited[i]:
            DFS(graph, i, visited)

def BFS(graph, v, visited):
    queue = deque([v])
    visited[v] = True
    while queue:
        v = queue.popleft()
        print(v, end=' ')
        for i in graph[v]:
            if not visited[i]:
                queue.append(i)
                visited[i] = True

DFS(graph, V, visited1)
print()
BFS(graph, V, visited2)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
const [N, M, V] = input[0].split(' ').map(Number);
const links = input.slice(1);

let graph = Array.from({ length: N + 1 }, () => []);
let visited1 = Array(N + 1).fill(false);
let visited2 = Array(N + 1).fill(false);
let result1 = [];
let result2 = [];

links.forEach((line) => {
  const [x, y] = line.split(' ').map(Number);
  graph[x].push(y);
  graph[y].push(x);
  graph[x].sort((a, b) => a - b);
  graph[y].sort((a, b) => a - b);
})

dfs(graph, visited1, V);
bfs(graph, visited2, V);
console.log(result1.join(' '));
console.log(result2.join(' '));

function dfs(graph, visited, v) {
  visited[v] = true;
  result1.push(v);
  for (let node of graph[v]) {
    if (!visited[node]) {
      dfs(graph, visited, node);
    }
  }
}

function bfs(graph, visited, v) {
  visited[v] = true;
  let queue = [v];
  while (queue.length) {
    const node = queue.shift();
    result2.push(node);
    for (let neighbor of graph[node]) {
      if (!visited[neighbor]) {
        queue.push(neighbor);
        visited[neighbor] = true;
      }
    }
  }
}